libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/useAdminInformationMessage.ts

```diff
@@ -0,0 +1,49 @@
+// (C) 2023 GoodData Corporation
+
+import { useEffect, useMemo, useState } from "react";
+import { ObjRef, objRefToString } from "@gooddata/sdk-model";
+import { stringUtils } from "@gooddata/util";
+import isNil from "lodash/isNil";
+
+const LOCAL_STORAGE_KEY_PREFIX = "gdc_share_dialog_admin_message_visible_";
+
+const createLocalStorageKey = (currentUserRef: ObjRef): string =>
+    LOCAL_STORAGE_KEY_PREFIX + stringUtils.simplifyText(objRefToString(currentUserRef));
+
+const getLocalStorageValue = <T>(key: string, defaultValue: T) => {
+    const item = localStorage.getItem(key);
+    const value = JSON.parse(item);
+
+    return isNil(value) ? defaultValue : value;
+};
+
+export const useLocalStorage = <T>(
+    key: string,
+    defaultValue: T,
+): [T, React.Dispatch<React.SetStateAction<T>>] => {
+    const [value, setValue] = useState<T>(() => getLocalStorageValue(key, defaultValue));
+
+    useEffect(() => {
+        localStorage.setItem(key, JSON.stringify(value));
+    }, [key, value]);
+
+    return [value, setValue];
+};
+
+interface IUseAdminInformationMessageStateReturnType {
+    isMessageVisible: boolean;
+    handleMessageClose: () => void;
+}
+
+export const useAdminInformationMessageState = (
```

Comment:
Is local storage a good place to store this information?
This means that on every new browser / domain, the admin will see this message again and again, it can be a bit annoying...

Reply:
That is exactly what we are going for here